### PR TITLE
TMDM-14042 Validate element's name of data model schema

### PR DESCRIPTION
**Pull request by sbliu**
**What is the current behavior?** (You should also link to an open issue here)
There missing element's name check of data model.

**What is the new behavior?**
Add rule to check element's name of data model.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/MetadataRepository.java [hunk 1]
```diff
@@ -166,6 +166,9 @@
 
     private final Map<String, Map<String, TypeMetadata>> entityTypes = new HashMap<String, Map<String, TypeMetadata>>();
 
+    // designed to element(entity without key info) name validation check
+    private final Map<String, Map<String, TypeMetadata>> entityTypes_withoutPk = new HashMap<String, Map<String, TypeMetadata>>();
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/MetadataRepository.java]: Please use same naming policy with existing code. 
Could name as `entityTypesWithoutPK`

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/MetadataRepository.java [hunk 2]
```diff
@@ -703,6 +721,14 @@ public void visitElement(XSDElementDeclaration element) {
             if (type.getKeyFields().isEmpty() && type.getSuperTypes().isEmpty()) {
                 Map<String, TypeMetadata> userEntityTypes = entityTypes.get(getUserNamespace());
                 if (userEntityTypes != null) {
+                    // record entity type without any key info
+                    Map<String, TypeMetadata> userEntityTypes_withoutPK = entityTypes_withoutPk.get(getUserNamespace());
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/MetadataRepository.java]: `userEntityTypesWithoutPK`

File: main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/ElementNameValidationRule.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package org.talend.mdm.commmon.metadata.validation;
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.commmon/src/main/java/org/talend/mdm/commmon/metadata/validation/ElementNameValidationRule.java]: Missing copyright.

**Comment by hbhong** [comment]
Should have JUnit for the new validator.

**Pull request merged**